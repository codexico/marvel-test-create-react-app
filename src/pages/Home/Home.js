import React from 'react';
import PropTypes from 'prop-types';

import Header from '../../components/Header/Header';
import Search from '../../components/Search/Search';
import HeroesList from '../../components/HeroesList/HeroesList';

import './Home.css';

function Home({ handleClick }) {
  return (
    <div className="Home">
      <Header />
      <Search />
      <HeroesList handleClick={handleClick} />
    </div>
  );
}

Home.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default Home;
