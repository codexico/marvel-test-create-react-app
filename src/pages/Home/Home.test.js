import React from 'react';
import { render } from '@testing-library/react';

import Home from './Home';

test('Component renders correctly', () => {
  const { asFragment } = render(<Home handleClick={() => {}} />);

  expect(asFragment()).toMatchSnapshot();
});
