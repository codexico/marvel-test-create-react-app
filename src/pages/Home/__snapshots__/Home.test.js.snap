// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="Header"
    >
      <svg
        title="Marvel - Search heroes"
      >
        logo.svg
      </svg>
      <h2
        class="Header-title"
      >
        EXPLORE O UNIVERSO
      </h2>
      <h3
        class="Header-subtitle"
      >
        Mergulhe no domínio deslumbrante de todos os personagens clássicos que você ama - e aqueles que você descobrirá em breve!
      </h3>
    </div>
    <div
      class="Search"
    >
      <form
        class="Search-form"
      >
        <div
          class="Search-inputGroup"
        >
          <label
            class="Search-label"
            for=""
          >
            <button
              class="Search-inputButton"
            >
              <svg
                class="Search-inputIcon"
              >
                search.svg
              </svg>
            </button>
            <input
              class="Search-input"
              placeholder="Procure por heróis"
            />
          </label>
        </div>
      </form>
    </div>
    <div
      class="HeroesList"
    />
  </div>
</DocumentFragment>
`;
